### PR TITLE
Remove incompatible version of openssl098

### LINKED ISSUE #3844: Remove incompatible version of openssl098
Run `brew unlink openssl098` when old version of openssl detected

See #3656 for more context

### PR DESCRIPTION
**Pull request by pkuczynski**
Fixes #3844
Uses #3855

### CODE REVIEW

File: scripts/functions/requirements/osx_brew [hunk 1]
```diff
@@ -49,17 +49,25 @@ requirements_osx_brew_lib_available()
...
+requirements_osx_brew_libs_error()
+{
+    rvm_warn "There were package ${1} errors, make sure to read the log.
+
+Try \`brew tap --repair\` and make sure \`brew doctor\` looks reasonable.
+
+Check Homebrew requirements https://github.com/Homebrew/homebrew/wiki/Installation"
+}
+
+}
```
Comment (mpapis) [inline scripts/functions/requirements/osx_brew]: double closing of the function?

File: scripts/functions/requirements/osx_brew [hunk 2]
```diff
@@ -290,6 +311,8 @@ requirements_osx_brew_libs_default()
 
   requirements_osx_brew_check_custom homebrew/dupes # for: zlib
 
+  undesired_check openssl098
```
Comment (mpapis) [inline scripts/functions/requirements/osx_brew]: not sure it should be in `custom` or rather in `before` or `define`? or maybe we should add `before_install`? or even a new step `uninstall`? - if `custom` is called always - just leave it :)

File: scripts/functions/requirements/osx_brew (hunk 1 above)
Comment (pkuczynski) [inline scripts/functions/requirements/osx_brew]: good catch! :)

File: scripts/functions/requirements/osx_brew (hunk 2 above)
Comment (pkuczynski) [inline scripts/functions/requirements/osx_brew]: Yeah, the `requirements_osx_brew_libs_default` is called in few places, that's why I put it there. Not sure if we should have a dedicated `uninstall` function. Possibly. Didn't want to invest too much of the time having RVM2 down the road...

**Pull request merged**